api/src/main/java/org/openmrs/module/bedmanagement/service/BedManagementService.java

```diff
@@ -0,0 +1,303 @@
+    BedDetails unAssignPatientFromBed(Patient patient);
+
+    /**
+     * Get Bed details by visit uuid
+     * @param visitUuid {@link String} visit uuid
+     * @return {@link BedDetails}
+     */
+    @Authorized(value = {"Get Beds", "Get Admission Locations"}, requireAll=true)
+    BedDetails getLatestBedDetailsByVisitUuid(String visitUuid);
+
+    /**
+     * Get all bed tags
+     *
+     * @return {@link List<BedTag>}
+     */
+    @Authorized(value = {"Get Tags"}, requireAll=true)
+    List<BedTag> getAllBedTags();
+
+    /**
+     * Get all wards
+     *
+     * @return {@link List<Location>}
+     */
+    @Authorized({"Get Admission Locations"})
+    List<Location> getAllWards();
+
+    /**
+     * Get wards by ward name
+     *
+     * @param name {@link String}
+     * @return {@link List<Location>}
+     */
+    @Authorized({"Get Admission Locations"})
+    List<Location> getWardsByName(String name);
+
+    /**
+     * Get ward by ward {@link Location} uuid
+     *
+     * @param uuid {@link String} location uuid
+     * @return {@link Location}
+     */
+    @Authorized({"Get Admission Locations"})
+    Location getWardByUuid(String uuid);
+
+    /**
+     * Get total number of beds in ward's room
+     *
+     * @param location {@link Location} ward's room
+     * @return {@link Long}
+     */
+    @Authorized(value = "Get Beds", requireAll=true)
+    Long getTotalBeds(Location location);
+
+    /**
+     * Get bedLocationMapping by bedId
+     *
+     * @param bedId {@link Integer} bed Id
+     * @return {@link BedLocationMapping}
+     */
+    @Authorized(value = "Get Beds", requireAll=true)
+    BedLocationMapping getBedLocationMappingByBedId(Integer bedId);
+
+    /**
+     * Get bedLocationMapping by locationUuid and row and column
+     *
+     * @param locationUuid {@link String} location Uuid
+     * @param row {@link Integer} bed row
+     * @param column {@link Integer} bed column
+     * @return {@link BedLocationMapping}
+     */
+    @Authorized(value = "Get Beds", requireAll=true)
+    BedLocationMapping getBedLocationMappingByLocationUuidAndRowColumn(String locationUuid, Integer row, Integer column);
+
+    /**
+     * Save / Update bedLocationMapping
+     *
+     * @param locationUuid {@link String} locationUuid
+     * @param row {@link Integer} row
+     * @param column {@link Integer} column
+     * @param bed {@link Bed}
+     * @return {@link BedLocationMapping}
+     */
+    @Authorized(value = "Edit Beds", requireAll=true)
+    BedLocationMapping saveBedLocationMapping(String locationUuid, Integer row, Integer column, Bed bed);
+
+    /**
+     * Save / Update ward
+     * @param uuid {@link String} location uuid, if null save new ward else update ward
+     * @param properties {@link SimpleObject} ward properties
+     * @return {@link Location}
+     */
+    @Authorized({"Edit Admission Locations", "Manage Locations"})
+    Location saveWard(String uuid, SimpleObject properties);
```

Comment:
Again, `SimpleObject` is an OMOD-side type to transact with the client-side, it shouldn't be used here.
As far as I understand a ward is nothing else than a location, so in the end this will just be equivalent to using directly the `LocationService`.